x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -434,10 +435,16 @@ else if (plan instanceof OrderBy) {
                 }
             }
 
-            return result;
+            return filterUnsupportedProjections(result);
```

Comment:
This part needs discussing. Hiding fields that exist from the resolution can easily become problematic.
A. SELECT * FROM table -> returns only supported results
B. SELECT unsupported_field FROM (SELECT * FROM TABLE) -> returns unknown field despite the field existing and knowing about it due to filtering

Who should win? A or B? I like the friendliness of A however I would argue that B should be the behaviour. That is if the users asks all fields, including those unsupported, we should do that. Otherwise we'll end with incorrect view of the worlds.
The same approach is used for dealing with mapping conflicts - we are aware of them however as long as the field is not used, things are good. Doing SELECT * on them though will bubble the exception.
That's because we don't want to resolve user conflicts, rather just be lazy about them.

Reply:
So, not positive I understand, sorry if getting it wrong, but:

> That is if the users asks all fields, including those unsupported, we should do that. Otherwise we'll end with incorrect view of the worlds.

We could try. (That's not what we do with ranges (or nested), though, for instance. It's simply unsupported/hidden.)
But the reason to hide it here, however, is:  what should an "old" driver do with an unsigned_long, even if the user explicitly asks for it. It'll still fail, but instead of doing it with an explicit server-generated error message, it'll fail with a more cryptic driver error.

We could maybe be more liberal if we (re)introduce restricting types based on client type (i.e. `isDriver()`)?

Additionally, that some apps do automatically run "SELECT *" (I'll share a log later).

If not a driver, the client could potentially choose (a) not to be versioned - i.e. skip communicating the version and so asking for server's version behaviour - or (b) choose a "recent enough" one.

But just to be sure:
> B. SELECT unsupported_field FROM (SELECT * FROM TABLE) -> returns unknown field despite the field existing and knowing about it due to filtering

This currently would still fail, assuming that `unsupported_field ` means in this case an `unsigned_long` field.

Reply:
I don't see an agreed conclusion to this discussion - please put it on the agenda for the next meeting.